tldraw/apps/tldraw-logseq/src/components/ActionBar/ActionBar.tsx

```diff
@@ -0,0 +1,31 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import * as React from 'react'
+import { observer } from 'mobx-react-lite'
+import type { Shape } from '~lib'
+import { App, useApp } from '@tldraw/react'
+import { Minimap } from '~components/Minimap'
+import { RedoIcon, UndoIcon } from '~components/icons'
+
+export const ActionBar = observer(function ToolBar(): JSX.Element {
+  const app = useApp<Shape>()
+  const undo = React.useCallback(() => {
```

Comment:
You can simply bind `onClick` with `app.api.undo` in https://github.com/logseq/logseq/pull/5742/files#diff-8087f14a3a56d73e3c412a20578c5be2299a46d9ea3673fd10a9e67478c63746R22

Reply:
I think I have fixed redo/undo in a new commit recently.

Reply:
Will retry this.

Reply:
I think I haven't fixed all the issues yet, lol..

Reply:
Looks to be working fine on my build. I think we can merge